### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by jakedt**

### CODE REVIEW

File: controllers/jobs.go [hunk 1]
```diff
@@ -43,6 +45,7 @@ func constructJobForMigration(managedDatabase *dba.ManagedDatabase, migration *d
 					RestartPolicy: corev1.RestartPolicyNever,
 				},
 			},
+			BackoffLimit: &noRetries,
```
[the remark this answers is not in the document]
Comment (jakedt) [reply controllers/jobs.go]: Maybe, we can add a `retryable` flag to the `databasemigration` in the future if we find use cases. The majority of the non-data migrations cannot be retried and an SRE should intervene if they fail.

File: controllers/manageddatabase_controller.go [hunk 2]
```diff
@@ -191,6 +191,8 @@ func (c *ManagedDatabaseController) reconcileMigrationJob(oneMigration migration
 				oneMigration.log.Info("Migration is complete")
 
 				// TODO: should we write the metric here or wait until cleanup?
+			} else if job.Status.Failed > 0 {
+				return fmt.Errorf("Migration job has errors: %s", job.Name)
```
[the remark this answers is not in the document]
Comment (jakedt) [reply controllers/manageddatabase_controller.go]: So many `if`s

[the remark this answers is not in the document]
Comment (jakedt) [reply controllers/manageddatabase_controller.go]: You're going to block until I remove it?

[the remark this answers is not in the document]
Comment (kleesc) [reply controllers/manageddatabase_controller.go]: I'm fine with checking only `Failed`. `JobCondition`s gives a bit more context though.

[the remark this answers is not in the document]
Comment (jakedt) [reply controllers/manageddatabase_controller.go]: Removed that commit so you'll approve.

**Pull request merged**